ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/KnownPeers.hs

```diff
@@ -174,16 +189,18 @@ jobPeerShare PeerSelectionActions{requestPeerShare}
         Right totalResults ->
           return $ Completion $ \st _ ->
            let peerResults = zip peers totalResults
-               -- Filter known-to-be ledger peers before adding them to known
+               -- Filter known-to-be big ledger peers before adding them to known
                -- peers. We to improve efficacy of Peer Sharing, because ledger
                -- peers are going to be hub nodes in the network (i.e. nodes
                -- with higher valency) they are going to be shared quite often,
                -- which means other smaller valency nodes won't be shared that
                -- much. So any chance we can have of filtering them from peer
                -- sharing results is welcome.
-               newPeers    = [ p | Right (PeerSharingResult ps) <- totalResults
+               newPeers    = takeNPeers maxAmount $
+                                 [ p | Right (PeerSharingResult ps) <- totalResults
                                  , p <- ps
-                                 , not (KnownPeers.member p (knownPeers st)) ]
+                                 , not (KnownPeers.member p (knownPeers st)) &&
+                                   Set.notMember p (PublicRootPeers.getBigLedgerPeers (publicRootPeers st))]
```

Comment:
```suggestion
                                 , not (KnownPeers.member p (knownPeers st))
                                 , Set.notMember p (PublicRootPeers.getBigLedgerPeers (publicRootPeers st))
                                 ]

Reply:
We're trying to help smaller SPOs to be discovered by both ledger peers and peer sharing, by not excluding also ledger peers.